File: backend/src/main/java/com/mystatslol/backend/DTO/MatchParticipantDetailDTO.java
package com.mystatslol.backend.DTO;

import com.mystatslol.backend.DTO.riotresponse.ParticipantRiotDTO;

public record MatchParticipantDetailDTO(
        String puuid,
        String riotIdGameName,
        String championName,
        int champLevel,
        boolean win,

        int summoner1Id,
        int summoner2Id,

        int[] items,

        int kills,
        int deaths,
        int assists,
       // double kdaRatio,

        int totalMinionsKilled,
      //  double csPerMinute,

        // Dégâts
        int totalDamageDealtToChampions
/*
        // Vision
        int visionScore,
        int controlWardsPlaced,

        // Runes
        int keystoneRuneId,
        int primaryStyle,
        int subStyle
        */
)
{
    public static MatchParticipantDetailDTO from(ParticipantRiotDTO p) {
        return new MatchParticipantDetailDTO(
                p.puuid(),
                p.riotIdGameName(),
                p.championName(),
                p.champLevel(),
                p.win(),
                p.summoner1Id(),
                p.summoner2Id(),
                new int[]{p.item0(), p.item1(), p.item2(), p.item3(), p.item4(), p.item5(), p.item6()},
                p.kills(),
                p.deaths(),
                p.assists(),
                p.totalMinionsKilled(),
                p.totalDamageDealtToChampions()
        );
    }
}

File: champion-api-go/models/MatchParticipantDetailDTO.go
package models

type MatchParticipantDetailDTO struct {
	Puuid          string `json:"puuid"`
	RiotIdGameName string `json:"riotIdGameName"`
	ChampionName   string `json:"championName"`
	ChampLevel     int    `json:"champLevel"`
	Win            bool   `json:"win"`

	Summoner1Id int `json:"summoner1Id"`
	Summoner2Id int `json:"summoner2Id"`

	Items []int `json:"items"`

	Kills   int `json:"kills"`
	Deaths  int `json:"deaths"`
	Assists int `json:"assists"`

	TotalMinionsKilled int `json:"totalMinionsKilled"`

	TotalDamageDealtToChampions int `json:"totalDamageDealtToChampions"`
}
